cli/hooks/webpack.js

```diff
@@ -0,0 +1,310 @@
+	}
+}
+
+function resolveAppcdClient(appcdPackagePath) {
+	const appcdClientPath = require.resolve('appcd-client', {
+		paths: [
+			path.join(appcdPackagePath, '..'),
+			path.join(appcdPackagePath, 'node_modules')
+		]
+	});
+	// eslint-disable-next-line security/detect-non-literal-require
+	return require(appcdClientPath).default;
+}
+
+class WebpackService {
+	constructor(options) {
+		this.client = new AppcdClient();
+		this.logger = options.logger;
+		const builder = options.builder;
+		const cli = options.cli;
+		this.projectPath = cli.argv['project-dir'];
+		this.platform = cli.argv.platform;
+		this.projectType = cli.tiapp.webpack.type;
+		this.shouldWatch = builder.deployType !== 'production' && !builder.buildOnly;
+		this.builder = builder;
+		this.forceRebuild = this.platform === 'ios' ? builder.forceCleanBuild : builder.forceRebuild;
+		this.jobIdentifier = this.generateJobIdentifier();
+		this.pluginVersion = '1.0.0';
+		this.basePath = `/webpack/${this.pluginVersion}`;
+		this.webpackStatusEmitter = new EventEmitter();
+		this.webpackStatus = { state: 'unknown' };
+	}
+
+	ensureDaemonIsRunning() {
+		return new Promise((resolve, reject) => {
+			this.logger.debug('Connecting to appcd ...');
+			this.client.connect({ startDaemon: true })
+				.once('connected', () => {
+					this.logger.info(`🖥  Web UI ready on http://localhost:1732${this.basePath}/web`);
+					resolve();
+				})
+				.once('error', reject);
+		});
+	}
+
+	ensureWebpackIsRunning() {
+		const options = {
+			identifier: this.jobIdentifier,
+			projectPath: this.projectPath,
+			deployType: this.builder.deployType,
+			watch: this.shouldWatch,
+			platform: this.platform,
+			projectType: this.projectType,
+			modules: this.builder.modules.map(m => m.id)
+		};
+
+		return new Promise((resolve, reject) => {
+			if (this.forceRebuild) {
+				this.startWebpack(options).then(resolve).catch(reject);
+				return;
+			}
+
+			this.client
+				.request({
+					path: `${this.basePath}/status/${options.identifier}`,
+				})
+				.once('response', response => {
+					this.webpackStatus = response;
+					if (this.webpackStatus.state === 'stopped' || this.webpackStatus.state === 'error') {
+						this.logger.debug('Webpack not running, starting new Webpack build.');
+						this.startWebpack(options).then(resolve).catch(reject);
+					} else {
+						if (!this.validateOptions(response, options)) {
+							this.logger.debug('Build options changed, restarting Webpack build.');
+							this.startWebpack(options).then(resolve).catch(reject);
+							return;
+						}
+						if (this.webpackStatus.state === 'ready') {
+							this.builder.tiSymbols = response.tiSymbols;
+						}
+						return resolve();
+					}
+				})
+				.once('error', error => {
+					if (error.status === 404) {
+						this.startWebpack(options).then(resolve).catch(reject);
```

Comment:
<!--
  0 failure: 
  2 warning:  cli/hooks/webpack..., cli/hooks/webpack...
  
  
  DangerID: danger-id-Danger;
  File: cli/hooks/webpack.js;
  Line: 200;
-->

- :warning: cli/hooks/webpack.js line 200 – Avoid using promises inside of callbacks. (promise/no-promise-in-callback)
- :warning: cli/hooks/webpack.js line 200 – Avoid using promises inside of callbacks. (promise/no-promise-in-callback)